### ISSUE TITLE
Energy normalization error on torch.fft.fft2 and torch.fft.ifft2 when the matrix size = [2048, 2048]

### ISSUE THREAD

**Issue by Pengju-Sheng** [pytorch/pytorch#169670]
### 🐛 Describe the bug

torch.fft.fft2 and torch.fft.ifft2 produces incorrect energy normalization when the input size is exactly (2048, 2048).
This issue only appears on Intel CPUs and does not occur on AMD CPUs or CUDA backends.

# minimal example
import torch as t

a = t.rand([2048, 2048])
print("The total energy of a:", t.sum(t.abs(a) ** 2))
b = t.fft.ifft2(t.fft.fft2(a))
print("The total energy of b:", t.sum(t.abs(b) ** 2))

### Versions

PyTorch version: 2.10.0.dev20250929+cu130
Is debug build: False
CUDA used to build PyTorch: 13.0
ROCM used to build PyTorch: N/A

OS: Could not collect
GCC version: (MinGW-W64 x86_64-ucrt-posix-seh, built by Brecht Sanders, r2) 15.2.0
Clang version: Could not collect
CMake version: version 4.1.2
Libc version: N/A

Python version: 3.13.7 (tags/v3.13.7:bcee1c3, Aug 14 2025, 14:15:11) [MSC v.1944 64 bit (AMD64)] (64-bit runtime)
Python platform: Windows-11-10.0.26200-SP0
Is CUDA available: True
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to:
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 5070 Ti Laptop GPU
Nvidia driver version: 581.29
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
'wmic' is not recognized as an internal or external command,
operable program or batch file.

Versions of relevant libraries:
[pip3] numpy==2.3.3
[pip3] torch==2.10.0.dev20250929+cu130
[pip3] torchvision==0.25.0.dev20250929+cu130
[conda] No relevant packages

cc @jianyuh @nikitaved @mruberry @walterddr @xwang233 @Lezcano @frank-wei @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10
